File: test/node/yarn/typescript-service/success-deployment-smoke/src/router.ts
import { IncomingMessage, ServerResponse } from 'node:http';
import { Configuration } from './configuration';
import { ZodError } from 'zod';
import { summarize } from './service';
export function createHandler(configuration: Configuration) {
  return (request: IncomingMessage, response: ServerResponse) => {
    const url = new URL(request.url ?? '/', 'http://localhost');
    let status = configuration.status;
    let type = 'text/plain; charset=utf-8';
    let body = configuration.label;
    if (status !== 503 && (url.pathname === '/api/summary' || configuration.mode === 'json')) {
      try {
        body = JSON.stringify(summarize(url.pathname === '/api/summary' ? url.searchParams.get('values') : null));
        type = 'application/json; charset=utf-8';
      } catch (error) {
        if (!(error instanceof ZodError)) throw error;
        status = 400; body = 'invalid-values';
      }
    }
    response.writeHead(status, {'content-type': type, 'content-length': Buffer.byteLength(body)});
    response.end(body);
  };
}

File: test/single-language/c/cmake/cpp-service/failure-health-rollback/src/router.cpp
#include "../include/router.hpp"
#include "../include/summary.hpp"
#include <stdexcept>
Response route_request(const Configuration &configuration, std::string_view target) {
    if (configuration.status == 503) return {503, "text/plain; charset=utf-8", configuration.label};
    auto separator = target.find('?');
    bool summary_route = target.substr(0, separator) == "/api/summary";
    if (summary_route || configuration.mode == "json") {
        try {
            auto query = summary_route && separator != std::string_view::npos ? target.substr(separator + 1) : "";
            return {200, "application/json; charset=utf-8", Summary::parse(query).json()};
        } catch (const std::invalid_argument &) { return {400, "text/plain; charset=utf-8", "invalid-values"}; }
    }
    return {200, "text/plain; charset=utf-8", configuration.label};
}
